config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/beans/Bundle.java

```diff
@@ -14,12 +14,13 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class Bundle {
     private final Map<String, Service> services = new HashMap<>();
     private final Map<String, Encass> encasses = new HashMap<>();
     private final Map<String, Policy> policies = new HashMap<>();
-    private final Map<String, Folder> folders = new HashMap<>();
+    private final Map<String, Folder> folders = new ConcurrentHashMap<>();
```

Comment:
why do we need concurrent map here?

Reply:
folders used to only be created from the PolicyAndFolderLoader. Since we can now make folders given a service path, folders can also be created in the ServiceLoader. Using a ConcurrentHashMap since the loaders are running in parallel